### ISSUE TITLE
Service locator is not registering services decorated with attribute "ServiceLocatorRegistration"

### ISSUE THREAD

**Issue by gofrogi** [Catel/Catel#1504]
since version > 5.9 such kind of registration doesn't work anymore.
`[ServiceLocatorRegistration(typeof(IPropertyVersionService), 
 ServiceLocatorRegistrationMode.Transient)]
    public class PropertyVersionService : IPropertyVersionService
    {
        ...
    } `
oddly enough, registration without second parameter works.

## Component
ServiceLocator

## Version of Library
5.11.2

## Version of OS(s) listed above with issue
Windows 10

## Steps to Reproduce
1. Create a interface ITestService
2. Create a class TestService : ITestService
3. Decorate the class TestService with Attribute [ServiceLocatorRegistration(typeof(ITestService), ServiceLocatorRegistrationMode.Transient]
4. anywhere in program.cs insert the code: 
```
ServiceLocator.Default.AutoRegisterTypesViaAttributes = true;
var test = ServiceLocator.Default.ResolveType<ITestService>();
```
5. start the program

## Expected Behavior
test should contain a object of type TestService

## Actual Behavior
UnregisteredServiceException or something alike.

**Comment by GeertvanHorrik** [comment]
Please provide a failing unit test so we can look into this.

**Comment by gouranga** [comment]
@GeertvanHorrik There you go [WpfApp1.zip](https://github.com/Catel/Catel/files/3979209/WpfApp1.zip). If you change nuget packages to 5.9.0 — everything works.

Also, something is wrong with registering types with conversions and filtering. For example I have a lot of Telerik controls and this line should exclude everything from Telerik.* namespace:
```
serviceLocator.RegisterTypesUsingAllConventions().ExcludeAllTypesOfNamespace("Telerik");
```
But it doesn't happening.

**Comment by GeertvanHorrik** [comment]
The reason is that the TypeCache is no longer aggressively loaded in advance:

https://github.com/Catel/Catel/blob/58d964c15281728972d9f1326048fdf665045663/src/Catel.Core/IoC/ServiceLocatorAutoRegistrationManager.cs#L152

The way to solve this with the current stable version is to call `TypeCache.InitializeTypes` before setting the property to `true`.

In the near future we will allow type cache initialization for the auto registration manager (makes sense when using the registration via attributes).

**Comment by GeertvanHorrik** [comment]
Fixed in Catel Build - 5.12.0-beta.25
